### PR TITLE
PIN-2972: added /eservices endpoint

### PR DESCRIPTION
**Pull request by AsterITA**

### CODE REVIEW

File: src/main/resources/interface-specification.yml [hunk 1]
```diff
@@ -750,6 +750,115 @@ paths:
               schema:
                 type: integer
               description: Time interval in milliseconds. Allowed requests will be constantly replenished during the interval. At the end of the interval the max allowed requests will be available
+  '/eservices':
+    get:
+      tags:
+        - gateway
+      summary: Retrieves EServices catalog
+      description: Retrieves EServices catalog
+      operationId: getEServicesCatalog
```
Comment (galales) [inline src/main/resources/interface-specification.yml]:
```suggestion
      operationId: getEServices
```

File: src/main/scala/it/pagopa/interop/apigateway/api/impl/GatewayApiServiceImpl.scala [hunk 2]
```diff
@@ -542,4 +542,34 @@ final case class GatewayApiServiceImpl(
       getKeyJWKfromKIdResponse[JWK](operationLabel)(getJWKByKid200)
     }
   }
+
+  override def getEServicesCatalog(limit: Int, offset: Int)(implicit
+    contexts: Seq[(String, String)],
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem],
+    toEntityMarshallerCatalogEServices: ToEntityMarshaller[CatalogEServices]
+  ): Route = {
+    val operationLabel = "Retrieving catalog of EServices"
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/apigateway/api/impl/GatewayApiServiceImpl.scala]:
```suggestion
    val operationLabel = "Retrieving EServices"
```

File: src/main/scala/it/pagopa/interop/apigateway/api/impl/ResponseHandlers.scala [hunk 3]
```diff
@@ -214,4 +214,12 @@ object ResponseHandlers extends AkkaResponses {
       case Failure(ex)              => internalServerError(ex, logMessage)
     }
 
+  def getEservicesCatalogResponse[T](logMessage: String)(
+    success: T => Route
+  )(result: Try[T])(implicit contexts: Seq[(String, String)], logger: LoggerTakingImplicit[ContextFieldsToLog]): Route =
+    result match {
+      case Success(s)                           => success(s)
+      case Failure(ex: OperationForbidden.type) => forbidden(ex, logMessage)
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/apigateway/api/impl/ResponseHandlers.scala]: can this happen?

File: src/main/scala/it/pagopa/interop/apigateway/service/impl/CatalogProcessServiceImpl.scala [hunk 4]
```diff
@@ -29,16 +29,23 @@ class CatalogProcessServiceImpl(invoker: CatalogProcessInvoker, api: EServiceApi
...
-  override def getEServices(producerId: UUID, attributeId: UUID, offset: Int, limit: Int)(implicit
-    contexts: Seq[(String, String)]
-  ): Future[EServices] = for {
+  override def getEServices(
+    name: Option[String] = None,
+    eServicesIds: Seq[UUID],
+    producersIds: Seq[UUID],
+    attributesIds: Seq[UUID],
+    agreementStates: Seq[AgreementState],
+    states: Seq[EServiceDescriptorState],
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/apigateway/service/impl/CatalogProcessServiceImpl.scala]: added parameters are not used in the call.
Are they useful? We can ignore them for now if not required

File: src/main/scala/it/pagopa/interop/apigateway/api/impl/ResponseHandlers.scala (hunk 3 above)
Comment (AsterITA) [reply src/main/scala/it/pagopa/interop/apigateway/api/impl/ResponseHandlers.scala]: > can this happen?

You're right, it shouldn't happen!

File: src/main/scala/it/pagopa/interop/apigateway/service/impl/CatalogProcessServiceImpl.scala (hunk 4 above)
Comment (AsterITA) [reply src/main/scala/it/pagopa/interop/apigateway/service/impl/CatalogProcessServiceImpl.scala]: > added parameters are not used in the call. Are they useful? We can ignore them for now if not required

I see, I forgot that they have a default and be omitted when not needed

**Review by galales — APPROVED** [review]

**Review by beetlecrunch — APPROVED** [review]

**Pull request merged**